backend/impl/db.go

```diff
@@ -34,47 +40,36 @@ func validateResult(r sql.Result, e error) error {
 	return nil
 }
 
-func updateUser(u UserArgs) error {
+type sqlDB struct {
+	db *sql.DB
+}
+
+func (s *sqlDB) updateUser(u UserArgs) error {
```

Comment:
What is the value of
func (s *sqlDB) updateUser(u UserArgs) error
vs
func updateUser(s *sqlDB, u UserArgs) error
?

Reply:
I'd ask you about that, the function was introduced by you. I only added a receiver to it.

Reply:
TL;DR: Readability.

Compare:
db.updateUser(args)
vs
updateUser(db, args)
The former locates you to a db definition. The latter gives you no idea which updateUser you're calling and where to look it for, considering there is a function with the similar name.